crates/biome_js_analyze/src/semantic_analyzers/nursery/no_nodejs_modules.rs

```diff
@@ -0,0 +1,87 @@
+use crate::globals::node::NODE_BUILTIN_MODULES;
+use crate::semantic_services::Semantic;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{JsCallExpression, JsImportDefaultClause, JsImportNamespaceClause};
+use biome_rowan::{declare_node_union, AstNode, TextRange};
+
+declare_rule! {
+    /// Forbid the use of Node.js builtin modules. Can be useful for client-side web projects that
+    /// do not have access to those modules.
+    ///
+    /// TODO(@anonrig): Add "allow" array option to allow specific modules.
+    ///
+    /// Source: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-nodejs-modules.md
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// import fs from "fs";
+    /// import path from "node:path";
+    /// ```
+    pub(crate) NoNodejsModules {
+        version: "next",
+        name: "noNodejsModules",
+        recommended: false,
+    }
+}
+
+impl Rule for NoNodejsModules {
+    type Query = Semantic<AnyJsImportLike>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let label_statement = ctx.query();
+
+        // Handle require() calls.
+        if let AnyJsImportLike::JsCallExpression(call) = label_statement {
```

Comment:
@ematipico This doesn't seem to be working correctly. Any suggestions?

Reply:
It does work, although `arguments` isn't the node you need. That's the node you're querying:

https://github.com/biomejs/biome/blob/3262574d41863c0406e85588e788cd3e55a199c2/xtask/codegen/js.ungram#L1625

You have to query `args()`, which is a **list**. You need to make sure it has **one** argument, and pick that first argument, and cast it to a string literal (because an argument can be anything). Here's some pseudo code: 

```rs
let first = args.iter().next()?.ok()?;
let string_literal = first
    .as_any_js_expression()?
    .as_any_js_literal_expression()?
    .as_js_string_literal_expression();
```

Reply:
Thank you!